docs/deployment/on_kubernetes.md

```diff
@@ -4,4 +4,112 @@
+```
+
+When running shell, you can use cmd `kubectl describe pod <podName>` to check if the information meets expectations.
+
+### Master
+
+Spark on Kubernetes config master by using a special format.
+
+`k8s://https://<k8s-apiserver-host>:<k8s-apiserver-port>`
+
+You can use cmd `kubectl cluster-info` to get api-server host and port.
+
+### Spark Image
+
+Kubernetes requires users to supply images that can be deployed into containers within pods.
+
+Spark also ships within a `./bin/docker-image-tool.sh` script that can be used to build and publish the Docker images to use
+with the K8s backend.
+
+Example usage is:
+
+```shell
+./bin/docker-image-tool.sh -r <repo> -t my-tag build
+./bin/docker-image-tool.sh -r <repo> -t my-tag push
+# To build docker image with specify openJdk 
+./bin/docker-image-tool.sh -r <repo> -t my-tag -b java_image_tag=<openjdk:${java_image_tag}> build
+# To build additional PySpark docker image
+./bin/docker-image-tool.sh -r <repo> -t my-tag -p ./kubernetes/dockerfiles/spark/bindings/python/Dockerfile build
+# To build additional SparkR docker image
+./bin/docker-image-tool.sh -r <repo> -t my-tag -R ./kubernetes/dockerfiles/spark/bindings/R/Dockerfile build
+```
+
+Of course, you can build self-made docker image by modifying Dockerfile and source code.
+
+### ServiceAccount
+
+When use Client mode to submit application, spark driver use the kubeconfig to access api-service to create and watch
+executor pods.
+
+When use Cluster mode to submit application, spark driver pod use serviceAccount to access api-service to create and
+watch executor pods.
+
+In both cases, you need to figure out whether you have the permissions under the corresponding namespace. You can use
+following cmd to create serviceAccount (You need to have the kubeconfig which have the create serviceAccount permission)
+.
+
+```shell
+# create serviceAccount
+kubectl create serviceaccount spark -n <namespace>
+# binding role
+kubectl create clusterrolebinding spark-role --clusterrole=edit --serviceaccount=<namespace>:spark --namespace=<namespace>
+```
+
+### Volumes
+
+As it known to us all, Kubernetes can use configurations to mount volumes into driver and executor pods.
+
+* hostPath: mounts a file or directory from the host node’s filesystem into a pod.
+* emptyDir: an initially empty volume created when a pod is assigned to a node.
+* nfs: mounts an existing NFS(Network File System) into a pod.
+* persistentVolumeClaim: mounts a PersistentVolume into a pod.
+
+Note: Please
+see [the Security section of this document](http://spark.apache.org/docs/latest/running-on-kubernetes.html#security) for
+security issues related to volume mounts.
+
+```shell
+--conf spark.kubernetes.driver.volumes.[VolumeType].[VolumeName].mount.path=<mount path>
+--conf spark.kubernetes.driver.volumes.[VolumeType].[VolumeName].mount.readOnly=<true|false>
+--conf spark.kubernetes.driver.volumes.[VolumeType].[VolumeName].mount.subPath=<mount subPath>
+```
```

Comment:
this (--conf) is still for the SparkPi example? Kyuubi has not supported such type for arg so far